core/api/workspace.josh.hcl

```diff
@@ -60,49 +60,55 @@ interface "workspace" {
 
     output "components" "[]ComponentDescription" {}
   }
-  
+
+  method "get-component-status" {
```

Comment:
I wonder if we should add this method. It seems like instead we should add specs and state to the results of `describe-components`. In that context, component types are dynamic, so spec and state are just strings interpreted by the particular controller for each component type. However, we have a method `describe-processes` which provides a nice decoded representation of process-specific fields.

The reason I think that this is important is because right now `get-component-status` is implemented by querying process state directly. That means that anybody who wants process-specific component state needs to call that API on-demand when it needs that information. It would be better if all this information is serialized in to state on `refresh`. This way we can have generic logic for invalidation, change management, subscriptions, etc. for all this state.

Reply:
I thought that there was some value in having some distinction between the description of a process and the minute-to-minute status and being able to query them separately. But I then merged the two concepts and it made stuff a lot simpler. So I think you're right.